src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
-    getLSKey('ConfigureRecordingSetPage', 'selectedRecordingSetIndex'),
-    -1,
-  );
+  const updatePaths = (listName: string, isBuiltIn: boolean) => {
+    const getFilePath = (metaDataState: MetadataState, listName: string, isBuiltIn: boolean) => {
+      let filePath = '';
 
-  const removeRecordingSet = async (setToDelete: RecordingSet): Promise<void> => {
-    const willDelete = confirm(`Are you sure you want to delete ${setToDelete.name}?`);
-    if (willDelete) {
-      const newSets = recordingSets.filter((x) => x !== setToDelete);
-      setRecordingSets(newSets);
+      const stateToPathDeltaEnglish = {
+        'oto-ini': join('external', 'delta_eng_ver5', 'mkototemp.ini'),
+        'list-preview': join('external', 'delta_eng_ver5', 'デルタ式engver5_reclist.txt'),
+        dvcfg: join('external', 'To be added later'),
+      };
 
-      if (recordingSets[selectedRecordingSetIndex] === setToDelete) {
-        setSelectedRecordingSetIndex(-1);
-      }
+      const stateToPathChinese = {
+        'oto-ini': join('external', 'z.cvvc_normal', 'oto.ini'),
+        'list-preview': join('external', 'z.cvvc_normal', 'Reclist.txt'),
+        dvcfg: join('external', 'To be added later'),
+      };
 
-      if (recordingProject) {
-        const willDeleteFiles = confirm(`Are you sure you want to delete the files also?`);
-        if (willDeleteFiles) {
-          const recordingSetPath = join(recordingProject.rootPath, setToDelete.name);
-          return deleteFolder(recordingSetPath);
+      const stateToCustListEnding = {
+        'oto-ini': join('external', 'oto.ini'),
+        'list-preview': join('external', 'Reclist.txt'),
+        dvcfg: join('external', 'voice.dvcfg'),
+      };
+      if (isBuiltIn) {
+        switch (listName) {
+          case 'デルタ式英語リストver5 (Delta English Ver. 5)':
+            filePath = join(filePath, stateToPathDeltaEnglish[metaDataState]);
+            break;
+          case 'Z式CVVC-Normal (Z Chinese CVVC - Normal)':
+            filePath = join(filePath, stateToPathChinese[metaDataState]);
+            break;
         }
+      } else {
+        filePath = join(listName, stateToCustListEnding[metaDataState]);
       }
-    }
-  };
 
-  const clearTemporaryItemsOnBack: typeof onBack = (e) => {
-    setCanWrite(false);
-    setChosenKey('C');
-    setChosenOctave(3);
-    setChosenName('');
-    rawSetChosenBuiltInList('');
-    rawSetChosenCustomListPath('');
-    setSelectedRecordingSetIndex(-1);
-    setProjectFile(DUMMY_PROJECT_FILE);
-    setShowingDetails(false);
-    setImmediate(() => onBack(e));
+      return filePath;
+    };
+
+    let isSubscribed = true;
+
+    const otoFilePath = getFilePath('oto-ini', listName, isBuiltIn);
+    const listFilePath = getFilePath('list-preview', listName, isBuiltIn);
+    const dvcfgFilePath = getFilePath('dvcfg', listName, isBuiltIn);
+
+    const listPromise = checkFileExistence(listFilePath);
```

Comment:
Still a lot of repetition here... Why not create an `async function checkMetadataFileExistence(metadataType: MetadataState, listName: string, isBuiltIn: boolean): Promise<boolean>`?